lib/chef/resource/powershell_package_source.rb

```diff
@@ -0,0 +1,162 @@
+# Author:: Tor Magnus Rakvåg (tm@intility.no)
+# Copyright:: 2018, Intility AS
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class Chef
+  class Resource
+    class PowershellPackageSource < Chef::Resource
+      resource_name "powershell_package_source"
+      provides(:powershell_package_source) { true }
+
+      description "Use the powershell_package_source resource to register a powershell package repository"
+      introduced "15.0"
+
+      property :name, String,
```

Comment:
A name_property of "name" is created for all resources out of the box. It would be more useful to make this an alternative name property like source_name so that someone could give a friendly name to the resource and still define the source name within the resource. This then requires updating any uses of new_resource.name to new_resource.source_name